benchmark/Streamly/Benchmark/Data/Parser.hs

```diff
@@ -161,6 +161,12 @@ split_ value =
             (PR.drainWhile (<= value))
         )
 
+{-# INLINE sliceSepBy #-}
+sliceSepBy :: MonadCatch m
+    => Int -> SerialT m Int -> m()
+sliceSepBy value = IP.parse (PR.sliceSepBy (>= value) (PR.many FL.drain
+                    (PR.satisfy (const True))))
```

Comment:
Indendation,
```
sliceSepBy value =
    IP.parse
        (PR.sliceSepBy (>= value) (PR.many FL.drain (PR.satisfy (const True))))
```
Why do you use `PR.many` there?
Two benchmarks of interest would be,
```
 IP.parse (PR.sliceSepBy (>= value) (PR.satisfy (const True))
```
to just see if it fuses. Should be compared to `takeWhile`
```
 IP.parse (PR.many FL.drain (PR.sliceSepBy ((== 0) . `mod` (value `div` 1000)) (PR.satisfy (const True))))
```
which is a little more complex use case.

Reply:
Maybe use,
```
PR.fromFold FL.drain
```
instead of,
```
PR.many FL.drain (PR.satisfy (const True))
```